File: homeassistant.ts
import { HA_API_KEY, HA_ENTITY } from './config'
import fetch from 'node-fetch'

export async function postLock(unlocked: boolean) {
  const service = unlocked ? 'unlock' : 'lock'

  await fetch(`http://171.66.3.236:8123/api/services/lock/${service}`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
      Authorization: `Bearer ${HA_API_KEY}`,
    },
    body: JSON.stringify({ entity_id: HA_ENTITY }),
  })
}

File: src/main/kotlin/dev/kdrag0n/quicklock/server/HomeAssistant.kt
package dev.kdrag0n.quicklock.server

import io.ktor.client.*
import io.ktor.client.engine.okhttp.*
import io.ktor.client.request.*

object HomeAssistant {
    private val client = HttpClient(OkHttp)

    suspend fun postLock(unlocked: Boolean, entityId: String) {
        val service = if (unlocked) "unlock" else "lock"

        val entity = Config.HA_ENTITIES[entityId]!!
        client.post("http://171.66.3.236:8123/api/services/lock/$service") {
            header("Content-Type", "application/json")
            header("Authorization", "Bearer ${Config.HA_API_KEY}")
            setBody("""{"entity_id": "${entity.haEntity}"}""")
        }
    }
}
